p4-fusion/thread_pool.h

```diff
@@ -39,16 +41,20 @@ class ThreadPool
 
 	std::condition_variable m_CV;
 
-	std::atomic<bool> m_ShouldStop;
+	std::mutex m_ShutdownMutex;
 	std::atomic<bool> m_HasShutDownBeenCalled;
 
 public:
 	ThreadPool(int size);
 	ThreadPool() = delete;
 	~ThreadPool();
 
-	void AddJob(Job function);
+	void AddJob(const Job& function);

```

Comment:
The correct signature here should be `Job&&`, as the caller is expected to transfer the ownership of the Job to the threadpool, not to let the thread pool borrow the Job (which is what is implied by `const Job&`). Right now, the implementation of `AddJob` is performing a copy into the deque structure, which would be replaced by a `std::move(function)`.